metadata-ingestion/src/datahub/ingestion/api/source_helpers.py

```diff
@@ -366,3 +377,128 @@ def _prepend_platform_instance(
         return [BrowsePathEntryClass(id=urn, urn=urn)] + entries
 
     return entries
+
+
+def auto_incremental_lineage(
```

Comment:
this file is getting large - we should consider splitting it up in a follow up PR